### PR TITLE
FV-353 Plausibilisieren beim Hochladen - 1_Knotenarmnummer

### PR DESCRIPTION
**Pull request by jan-niklasml**
# Description

Adds validation of knotenarmnummer, when files are uploaded.

# Issue References

FV-353
 
# Definition of Done

- [x] Acceptance criteria are met
- [x] Testing is done (unit-tests, DEV-environment)
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification, etc.)

## Summary by CodeRabbit

* **Bug Fixes**
  * Snackbar now renders multi-line messages correctly and avoids stray carriage characters.
  * CSV upload validation improved: mapping plausibility checked, duplicate uploads per item detected and reported together, and processing counts adjusted for duplicates.

* **Tests**
  * Added unit tests covering multi-line snackbar formatting and edge cases (empty/undefined input).

[Review Change Stack](https://app.coderabbit.ai/change-stack/it-at-m/dave-selfservice-portal/pull/210?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/components/common/TheSnackbar.vue [hunk 1]
```diff
@@ -41,6 +46,10 @@ const color = ref(Levels.INFO);
 
 const snackbarStore = useSnackbarStore();
 
+const formattedSnackbarTextPart2 = computed(() =>
+  snackbarTextPart2.value?.split("\n")
+);
```
[the remark this answers is not in the document]
Comment (boal) [reply frontend/src/components/common/TheSnackbar.vue]: Die vorgeschlagenen Codeänderung zum save split implementieren und ggf. unittesten.

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue [hunk 2]
```diff
@@ -695,11 +697,41 @@ function readFiles() {
...
           snackbarStore.showError(
-            `Die Datei ${myFile.name} konnte keinem Knotenarm zugeordnet werden. Metadaten prüfen`
+            `Die Datei ${myFile.name} konnte keinem Knotenarm zugeordnet werden. Knotenarmnummer prüfen!`
           );
         }
+
+        // Plausibilisierung: Wurden mehrere Dateien mit dieser Knotenarmnummer hochgeladen?
+        if (knotenarmeWithUploadedFiles.has(knotenarmnummerOfCsv)) {
+          if (
+            !errorTextKnotenarmnummer.includes(
```
Comment (boal) [inline frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Den Code in der Prüfung des If-Statements ggf in eine eigene Methode auslagern die einen aussagekräftigen Namen und einen boolean zurückgibt.

Zusätzlich ggf unittesten

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue [hunk 3]
```diff
@@ -695,11 +697,41 @@ function readFiles() {
           /\r\n|\n/
         );
         const knotenarmnummerOfCsv: number = getKnotenarmnummerOfCsv(csv);
-        if (knotenarmnummerOfCsv === 0) {
+        // Plausibilisierung: Kann die Datei einem Knotenarm der Zählung zugeordnet werden?
+        if (
+          !zaehlung.value.knotenarme.some(
+            (zaehlungArm: KnotenarmDTO) =>
```
Comment (boal) [inline frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Den Code in der Prüfung des If-Statements ggf in eine eigene Methode auslagern die einen aussagekräftigen Namen und einen boolean zurückgibt.

Zusätzlich ggf unittesten

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue [hunk 4]
```diff
@@ -695,11 +697,41 @@ function readFiles() {
...
+          )
+        ) {
           snackbarStore.showError(
-            `Die Datei ${myFile.name} konnte keinem Knotenarm zugeordnet werden. Metadaten prüfen`
+            `Die Datei ${myFile.name} konnte keinem Knotenarm zugeordnet werden. Knotenarmnummer prüfen!`
           );
         }
+
+        // Plausibilisierung: Wurden mehrere Dateien mit dieser Knotenarmnummer hochgeladen?
+        if (knotenarmeWithUploadedFiles.has(knotenarmnummerOfCsv)) {
```
Comment (boal) [inline frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Die Prüfung sollte nacheiander stattfinden. Somit sind die if-Statement mit einem else an das vorhherige If-Statement anzuschließen.

**Review by boal — CHANGES_REQUESTED** [review]

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue (hunk 3 above)
Comment (jan-niklasml) [reply frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Wurde in Methode ausgelagert und getestet

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue (hunk 4 above)
Comment (jan-niklasml) [reply frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Umgesetzt

File: frontend/src/components/zaehlung/form/KnotenLageForm.vue (hunk 2 above)
Comment (jan-niklasml) [reply frontend/src/components/zaehlung/form/KnotenLageForm.vue]: Wurde vereinfacht

File: frontend/src/components/common/TheSnackbar.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (jan-niklasml) [reply frontend/src/components/common/TheSnackbar.vue]: Save-Split wurde implementiert und getestet

**Review by boal — APPROVED** [review]
